### PR TITLE
[Feat] #7118, #7778: add popup, change manager name, provide optional responsible persons

### PR DESCRIPTION
**Pull request by yuliiakaras**
[#7778](https://github.com/ita-social-projects/GreenCity/issues/7778)
[#7118](https://github.com/ita-social-projects/GreenCity/issues/7118)

### CODE REVIEW

File: src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts [hunk 1]
```diff
@@ -703,7 +701,7 @@ export class UbsAdminOrderComponent implements OnInit, OnDestroy, AfterContentCh
       this.isFormResetted = true;
     } else {
       exportDetaisFields.forEach((el) => exportDetails.get(el).setValidators([Validators.required]));
-      responsiblePersonNames.forEach((el) => responsiblePersons.get(el).setValidators([Validators.required]));
+      responsiblePersons.get('responsibleCaller')?.setValidators([Validators.required]);
     }
     this.statusCanceledOrDone();
   }
```
Comment (Lokankara) [inline src/app/ubs/ubs-admin/components/ubs-admin-order/ubs-admin-order.component.ts:707]: Expected 1 arguments, but got 0:
`this.destroy$.next();`

File: src/app/ubs/ubs-admin/components/ubs-admin-responsible-persons/ubs-admin-responsible-persons.component.ts [hunk 2]
```diff
@@ -66,6 +68,45 @@ export class UbsAdminResponsiblePersonsComponent implements OnInit, OnDestroy, O
...
   }
 
+  isFieldOptional(controlName: string): boolean {
+    const requiredFields = ['responsibleCaller'];
+
+    if (this.orderStatus !== 'ADJUSTMENT') {
+      return false;
+    }
+
+    if (requiredFields.includes(controlName)) {
```
Comment (Lokankara) [inline src/app/ubs/ubs-admin/components/ubs-admin-responsible-persons/ubs-admin-responsible-persons.component.ts]: 'if' statement can be simplified

File: src/app/ubs/ubs-admin/components/ubs-admin-responsible-persons/ubs-admin-responsible-persons.component.ts [hunk 3]
```diff
@@ -66,6 +68,45 @@ export class UbsAdminResponsiblePersonsComponent implements OnInit, OnDestroy, O
...
+      {
+        translate: 'responsible-persons.driver',
+        formControlName: 'responsibleDriver',
+        responsiblePersonsArray: this.allDrivers
+      }
+    ];
+  }
+
   ngOnDestroy(): void {
     this.destroy$.next();
```
Comment (Lokankara) [inline src/app/ubs/ubs-admin/components/ubs-admin-responsible-persons/ubs-admin-responsible-persons.component.ts]: Expected 1 arguments, but got 0.

File: src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-select/table-cell-select.component.ts [hunk 4]
```diff
@@ -114,13 +115,22 @@ export class TableCellSelectComponent implements OnInit {
...
-      this.checkIfStatusConfirmed();
-    } else if (this.nameOfColumn === 'orderStatus' && (this.newOption === 'Canceled' || this.newOption === 'Скасовано')) {
-      this.openCancelPopUp();
-    } else if (this.nameOfColumn === 'orderStatus') {
+  saveClick(): void {
+    if (this.nameOfColumn !== 'orderStatus') {
       this.save();
+      return;
+    }
+    const isCancelOption = ['Canceled', 'Скасовано'].includes(this.newOption);
```
Comment (Lokankara) [inline src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-select/table-cell-select.component.ts]: By using the `enums OrderStatusEn and OrderStatusUa`  and `isStatusOption` to categorize status options, the logic becomes more scalable and easier to maintain 
```
  private isStatusOption(...options: string[]): boolean {
    return options.includes(this.newOption);
  }
```

File: src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-select/table-cell-select.component.ts [hunk 5]
```diff
@@ -114,13 +115,22 @@ export class TableCellSelectComponent implements OnInit {
...
+    const isConfirmOption = ['Сформовано', 'Formed', 'Confirmed', 'Підтверджено', 'Привезе сам', 'Brought by himself'].includes(
+      this.newOption
+    );
+
+    if (isConfirmOption) {
+      this.openConfirmPopUp();
+    } else if (isCancelOption) {
+      this.openCancelPopUp();
+    } else if (this.checkStatus && this.showPopUp) {
+      this.checkIfStatusConfirmed();
```
Comment (Lokankara) [inline src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-select/table-cell-select.component.ts]: we can simplify this condition by not creating `checkIfStatusConfirmed`.
```
const isConfirmedOption = this.isStatusOption(OrderStatusEn.CONFIRMED, OrderStatusUa.CONFIRMED);
...

} else if (this.checkStatus && this.showPopUp && !isConfirmedOption) {
    this.openPopUp();
  } else {
    this.save();
  }
```

**Review by Lokankara — APPROVED** [review]

File: src/app/ubs/ubs-admin/components/ubs-admin-responsible-persons/ubs-admin-responsible-persons.component.spec.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (Lokankara) [inline src/app/ubs/ubs-admin/components/ubs-admin-responsible-persons/ubs-admin-responsible-persons.component.spec.ts:1]: 'async' is deprecated, use waitForAsync()

**Review by Lokankara — APPROVED** [review]

**Review by kovalsofiia1 — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Pull request merged**